src/bp/ui-studio/src/web/reducers/flows.ts

```diff
@@ -637,15 +637,37 @@ reducer = reduceReducers(
       },
 
       [requestUpdateFlowNode]: (state, { payload }) => {
+        payload = Array.isArray(payload) ? payload : [{ ...payload, id: payload.id ?? state.currentFlowNode }]
         const currentFlow = state.flowsByName[state.currentFlow]
-        const currentNode = _.find(state.flowsByName[state.currentFlow].nodes, { id: state.currentFlowNode })
-        const needsUpdate = name => name === (currentNode || {}).name && payload.name
+        const nodesToUpdate = payload.map(node =>
+          _.find(currentFlow.nodes, {
+            id: node.id
+          })
+        )
+
+        // Prevent errors by doing nothing if any nodes are undefined
+        if (nodesToUpdate.filter(node => !node).length) {
+          return state
+        }
+
+        // Find the replacement name, if there is any
+        const needsUpdate = name => {
```

Comment:
Can you change this function name since `needsUpdate` does not represent what it does anymore?